ext/calendar/tests/gh16235.phpt

```diff
@@ -0,0 +1,13 @@
+--TEST--
+GH-16235 (jdtogregorian overflow on argument)
+--EXTENSIONS--
+calendar
+--SKIPIF--
+<?php if (PHP_INT_SIZE != 8) die("skip this test is for 64bit platform only"); ?>
+--FILE--
+<?php
+jdtogregorian(92233658792494800);
```

Comment:
I sugest to test the smallest positive value which would overflow (on Windows, `536838867` is the smallest positive value which is rejected on x64 and x86; and that's still good enough for more than a million years).